pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -548,10 +547,25 @@ func (nim *nodeInfoManager) installDriverToCSINode(
 		return nil
 	}
 
+	var maxLimit *int32
+	if attachVolumeLimitEnabled && maxAttachLimit > 0 {
+		if maxAttachLimit > math.MaxInt32 || maxAttachLimit < 0 {
+			klog.Errorf("Invalid attach limit value %d cannot be added to CSINode object for %q", maxAttachLimit, driverName)
+		} else {
+			m := int32(maxAttachLimit)
+			maxLimit = &m
+		}
+	} else {
+		klog.V(4).Infof("Skipping adding attach limit to CSINode object for %q", driverName)
+	}
+
 	// Append new driver
 	driverSpec := storagev1beta1.CSINodeDriver{
-		Name:         driverName,
-		NodeID:       driverNodeID,
+		Name:   driverName,
+		NodeID: driverNodeID,
+		Allocatable: storagev1beta1.VolumeLimits{
+			Count: maxLimit,
+		},
```

Comment:
We should set this only when attach limit feature gate is enabled.

Reply:
Fixed